src/doctor.ts

```diff
@@ -0,0 +1,66 @@
+import { getProtocolInterface } from "./deps.ts";
+
+type RuntimeVersion = {
+  name: string;
+  current: string;
+  minimum?: string;
+  error?: {
+    message: string;
+  };
+};
+
+const getHostedDenoRuntimeVersion = async (): Promise<{
+  minimum?: string;
+  message?: string;
+  error?: { message: string };
+}> => {
+  try {
+    const metadataURL = "https://api.slack.com/slackcli/metadata.json";
+    const response = await fetch(metadataURL);
+    if (!response.ok) {
+      throw new Error("Failed to collect upstream CLI metadata");
+    }
+    const metadata = await response.json();
+    const version = metadata?.["deno-runtime"]?.releases[0]?.version;
+    if (!version) {
+      throw new Error("Failed to find the minimum Deno version");
```

Comment:
Let's add the `metadata.cli` response payload as part of the error too.

Reply:
Good call. I think this is useful, even if no obvious remediation exists

![metadata-response](https://github.com/slackapi/deno-slack-hooks/assets/18134219/e2eedbba-1ee5-46db-826b-83397b20146a)

Reply:
Hmm, thoughts on maybe pumping the specific error to the debug log and not messing up the delicate doctor template? We may have to use the `protocol`'s differing `respond` vs. `log` methods to enable the CLI to distinguish between the two.